### PR TITLE
Update for new wlroots

### PR DESCRIPTION
**Pull request by valpackett**
+ some cleanup

### CODE REVIEW

File: tablecloth/decoration.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ void main()
 
     auto Context::render_decorations(View& view, RenderData& data) -> void
     {
-      if (view.maximized || view.wlr_surface == nullptr) {
+      if (view.maximized || view.wlr_surface == nullptr || !view.deco.is_visible()) {
```
Comment (topisani) [inline tablecloth/decoration.cpp:240]: I think in most cases shadows should be drawn even when no decorations are used

File: tablecloth/decoration.cpp [hunk 2]
```diff
@@ -254,10 +254,8 @@ void main()
 
       draw_shadow(box, view.rotation, 0.5 * data.alpha, view.deco.shadow_radius(), view.deco.shadow_offset());
 
-      if (!view.deco.is_visible()) return;
-
       wlr::box_t rotated;
-      wlr_box_rotated_bounds(&box, view.rotation, &rotated);
+      wlr_box_rotated_bounds(&rotated, &box, view.rotation);
```
Comment (topisani) [inline tablecloth/decoration.cpp:258]: did wlroots change the order?

File: tablecloth/render.cpp [hunk 3]
```diff
@@ -342,17 +342,8 @@ namespace cloth::render {
       double view_y = (double) (output_box->height - view_box.height) / 2 + output_box->y;
       view.move(view_x, view_y);
 
-      if (has_standalone_surface(view) &&
```
Comment (topisani) [inline tablecloth/render.cpp:345]: why?

**Review by topisani — CHANGES_REQUESTED** [review]
When updating to the new wlroots, all changes from rootston should be ported as well

Not sure i want to merge this until that part is done too.

File: tablecloth/decoration.cpp (hunk 2 above)
Comment (valpackett) [reply tablecloth/decoration.cpp:258]: yes exactly

File: tablecloth/render.cpp (hunk 3 above)
Comment (valpackett) [reply tablecloth/render.cpp:345]: This is ported from rootston. `wlr_output_set_fullscreen_surface` no longer exists

File: tablecloth/decoration.cpp (hunk 1 above)
Comment (valpackett) [reply tablecloth/decoration.cpp:240]: GTK draws its own shadows! This results in very ugly double shadows!

Comment (topisani) [reply tablecloth/decoration.cpp:240]: alright, makes sense

**Comment by valpackett** [comment]
Ported some more rootston changes, fixed HiDPI scaling and damage.

I see two bugs that aren't present in rootston:

- GTK menus (top menubar, right click context) are unusable, they get closed as soon as they open
- Firefox popovers (right click context menu, address suggestions, etc. — note: not xdg popups, they're implemented as subsurfaces right now) just block Firefox from receiving any further input (except window resizes) :(

**Pull request closed without merging**